src/Components/Aspire.Azure.Messaging.EventHubs/EventHubsComponent.cs

```diff
@@ -46,17 +46,21 @@ protected static string GetNamespaceFromSettings(AzureMessagingEventHubsSettings
 
         try
         {
-            // extract the namespace from the connection string or qualified namespace
-            var fullyQualifiedNs = string.IsNullOrWhiteSpace(settings.FullyQualifiedNamespace) ?
-                EventHubsConnectionStringProperties.Parse(settings.ConnectionString).FullyQualifiedNamespace :
-                new Uri(settings.FullyQualifiedNamespace).Host;
+            // Extract the namespace from the connection string or qualified namespace
+            ns = string.IsNullOrWhiteSpace(settings.FullyQualifiedNamespace)
+                ? EventHubsConnectionStringProperties.Parse(settings.ConnectionString).Endpoint.Host
+                : new Uri(settings.FullyQualifiedNamespace).Host;
 
-            ns = fullyQualifiedNs[..fullyQualifiedNs.IndexOf('.')];
+            // This is likely to be similar to {yournamespace}.servicebus.windows.net.
+            if (ns.EndsWith(".servicebus.windows.net", StringComparison.OrdinalIgnoreCase))
```

Comment:
What about other clouds like China and government? Will this work for them?

cc @jsquire

Reply:
This will differ between clouds.   I believe China is `.servicebus.chinacloudapi.cn` and US Gov is `servicebus.usgovcloudapi.net`, though I'm not 100% on either.   We don't test in some of the more specialized clouds, so we may see even greater variation.

Reply:
Is the suffix important here?  If we're just trying to trim off the namespace name from the FQNS, couldn't we just do a `Contains('.')` check and, if so, trim?

Reply:
The code before that was getting the first `.` but if you try using an IP address then you get the first number. And if you had "localhost" then it would simply fail because there was no ``. This code is used to create a Blob container

```c#
            // Do we have a container name provided in the settings?
            if (string.IsNullOrWhiteSpace(settings.BlobContainerName))
            {
                // If not, we'll create a container name based on the namespace, event hub name and consumer group
                settings.BlobContainerName = $"{ns}-{settings.EventHubName}-{consumerGroup}";
                shouldTryCreateIfNotExists = true;
            }
```

Maybe we could just omit this part from the name, and use some random value to prevent collisions:

Reply:
ah... fair point.   I hadn't considered the IP address scenario.    If we're looking to create a blob container, maybe we could just strip punctuation and force lowercase?   I don't recall exactly what the character limit is for those, but I think its enough that may work.     If not, maybe using the first 10-15 characters and adding a random suffiX

Reply:
I ended up checking for `.servicebus` and take the first part, like before but better (cn). Then in all other cases I generate a random 12 chars prefix, and it will still be`{random12}-{settings.EventHubName}-{consumerGroup}`. If I were to use "localhost" for instance I am worried that it could conflict with other instance on different ports.